jsx-factory.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import('./lib/core.js').Element} Element
+ * @typedef {import('./lib/core.js').Root} Root
+ * @typedef {import('./lib/core.js').HResult} HResult
+ * @typedef {import('./lib/core.js').HArrayChild} HArrayChild
+ * @typedef {import('./lib/core.js').HProperties} HProperties
+ * @typedef {import('./lib/core.js').HPropertyValue} HPropertyValue
+ * @typedef {import('./lib/core.js').HStyle} HStyle
+ * @typedef {import('./lib/core.js').core} Core
+ *
+ * @typedef {Record<string, unknown>} JSXProps
+ * @typedef {(props: JSXProps) => HResult } HFn
+ */
+
+import {h as hast} from './index.js'
```

Comment:
* This doesn’t support SVG
* Instead of creating another factory, and an HTML and SVG function (and in the future mathml), I think I now prefer to just add this to the regular `h`; also the affect on the docs gets a bit complex with a separate `/jsx-factory`?

Reply:
Ok, so to confirm, we'll enhance the core [`h()`](https://github.com/cowboyd/hastscript/blob/functional-jsx-elements/lib/core.js#L54) function to accept `null`, `string`, or `function`. Do you want me to document this feature? It seems like it's not very useful when you are using hypersrcipt because you can just call the function yourself. I.e. why say: 

```js
h(DLEntry, { key: "x", value: "y" })
```

When you could just as easily say:

```js
DLEntry({ key: "x", value: "y" });
```

There might be a reason to invoke it with the former, but I can't think of it. What if I just note on the `@param` declaration "accepts a function in order to support JSX"

Reply:
React and preact are the same? `React.createElement(DlEntry, {key, value})` vs `DlEntry({key, value})` vs <DlEntry key={key} value={value})>`. The main reason is that you don’t have to create 2 wrappers?

Reply:
I agree that this is simpler. I'm just asking how to explain to users who _don't_ use JSX why this is also something that the `h` function can do:

```ts
h(DLEntry, { key: "x", value: "y" })
```

If I wasn't using JSX, I would be like "why is that ever useful", so I was suggesting a brief note in the jsdoc that this is for JSX support.